### PR TITLE
[FINERACT-1678] Exclude loan IDs from execution context parameter name fix

### PR DESCRIPTION
**Pull request by taskain7**
## Description

fixing the paratemer name for `Loan COB parameter`

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyLoanLockTasklet.java [hunk 1]
```diff
@@ -65,7 +65,7 @@ public RepeatStatus execute(@NotNull StepContribution contribution, @NotNull Chu
             loanIds = Collections.emptyList();
         } else {
             loanIds = new ArrayList<>(loanRepository.findAllNonClosedLoansBehindOrNullByMinAndMaxLoanId(loanCOBParameter.getMinLoanId(),
-                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE)));
+                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE).minusDays(1)));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyLoanLockTasklet.java]: why -1 day?

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBPartitioner.java [hunk 2]
```diff
@@ -66,34 +69,24 @@ private Map<String, ExecutionContext> getPartitions(int partitionSize, Set<Busin
...
         }
         return partitions;
     }
 
     private void createNewPartition(Map<String, ExecutionContext> partitions, int partitionIndex,
-            Set<BusinessStepNameAndOrder> cobBusinessSteps) {
+            Set<BusinessStepNameAndOrder> cobBusinessSteps, List<Long> loanIds) {
         ExecutionContext executionContext = new ExecutionContext();
         executionContext.put(LoanCOBConstant.BUSINESS_STEPS, cobBusinessSteps);
+        executionContext.put(LoanCOBConstant.LOAN_COB_PARAMETER, new LoanCOBParameter(loanIds.get(0), loanIds.get(loanIds.size() - 1)));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanCOBPartitioner.java:89]: why -1 day?

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java [hunk 3]
```diff
@@ -49,7 +49,7 @@ public void beforeStep(@NotNull StepExecution stepExecution) {
             loanIds = Collections.emptyList();
         } else {
             loanIds = loanRepository.findAllNonClosedLoansBehindOrNullByMinAndMaxLoanId(loanCOBParameter.getMinLoanId(),
-                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE));
+                    loanCOBParameter.getMaxLoanId(), ThreadLocalContextUtil.getBusinessDateByType(BusinessDateType.COB_DATE).minusDays(1));
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java]: why -1 day?

Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java:52]: Also this one is not correct as it is ignoring any already locked loans (locked by INLINE COB for example)

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java [hunk 4]
```diff
@@ -91,8 +91,7 @@ public interface LoanRepository extends JpaRepository<Loan, Long>, JpaSpecificat
     String FIND_ALL_NON_CLOSED_LOANS_BEHIND_OR_NULL_BY_LOAN_IDS = "select loan.id, loan.lastClosedBusinessDate from Loan loan where loan.id IN :loanIds and loan.loanStatus in (100,200,300,303,304) and (loan.lastClosedBusinessDate < :cobBusinessDate or "
             + "loan.lastClosedBusinessDate is null)";
 
-    String FIND_ALL_NON_CLOSED_LOANS_BEHIND_OR_NULL_BY_MIN_AND_MAX_LOAN_ID = "select loan.id from Loan loan where loan.id BETWEEN :minLoanId and :maxLoanId and loan.loanStatus in (100,200,300,303,304) and (loan.lastClosedBusinessDate < "
-            + ":cobBusinessDate or loan.lastClosedBusinessDate is null)";
+    String FIND_ALL_NON_CLOSED_LOANS_BEHIND_OR_NULL_BY_MIN_AND_MAX_LOAN_ID = "select loan.id from Loan loan where loan.id BETWEEN :minLoanId and :maxLoanId and loan.loanStatus in (100,200,300,303,304) and (:cobBusinessDate = loan.lastClosedBusinessDate or loan.lastClosedBusinessDate is NULL)";
```
Comment (adamsaghy) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java]: the naming is confusing, it is not filtering by it is behind or not... it checks whether the date is matching

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java (hunk 3 above)
Comment (adamsaghy) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java]: It is ignoring if the loan is already locked and maybe failed...maybe it failed on the very first execution and the closed business date is still NULL, but it failed 2 days ago... it is edge case, but we need to be careful here...(probably not cause any issues in the other hand...)

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyLoanLockTasklet.java (hunk 1 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/ApplyLoanLockTasklet.java]: moved the `NUMBER_OF_DAYS_BEHIND` to `LoanCOBConstant` and use it here as well

File: fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java (hunk 3 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/cob/loan/LoanItemReader.java]: moved the `NUMBER_OF_DAYS_BEHIND` to `LoanCOBConstant` and use it here as well

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java (hunk 4 above)
Comment (taskain7) [reply fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/LoanRepository.java]: renamed

**Pull request merged**